src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -189,7 +189,7 @@ var TestRenderer = ReactFiberReconciler({
 
   useSyncScheduling: true,
 
-  getPublicInstance(inst) {
+  getPublicInstance(inst : Instance | TextInstance) {
```

Comment:
Ah, I see. Just put `getPublicInstance(inst : Instance | TextInstance) : *` then for "inferred". That way if createNodeMock gets a proper type in the future, it'll get caught.